### ISSUE TITLE
UTFDataFormatException happens when read seariazble

### ISSUE THREAD

**Issue by duoduo12** [OpenHFT/Chronicle-Map#183]
chronicle map Version 3.17.1, chronicle bytes 2.17.20

We have a chronicle map as a concurrent hash map, string as key and a seriazable class instance as value. The map is stateful cache for our app, and have lots of updates on the properties of the value instance. Generally it works well, but after serveral days it will throw the following exception, any sugguestion? Thanks

java.lang.RuntimeException: java.io.UTFDataFormatException
        at net.openhft.chronicle.hash.serialization.impl.SerializableReader.read(SerializableReader.java:45)
        at net.openhft.chronicle.hash.serialization.impl.SerializableReader.read(SerializableReader.java:36)
        at net.openhft.chronicle.hash.serialization.impl.SerializableReader.read(SerializableReader.java:31)
        at net.openhft.chronicle.map.VanillaChronicleMap.searchValue(VanillaChronicleMap.java:628)
        at net.openhft.chronicle.map.VanillaChronicleMap.tieredValue(VanillaChronicleMap.java:561)
        at net.openhft.chronicle.map.VanillaChronicleMap.optimizedGet(VanillaChronicleMap.java:513)
        at net.openhft.chronicle.map.VanillaChronicleMap.get(VanillaChronicleMap.java:462)
        at java.util.concurrent.ConcurrentMap.getOrDefault(ConcurrentMap.java:80)
...
Caused by: java.io.UTFDataFormatException
        at java.io.ObjectInputStream$BlockDataInputStream.readUTFSpan(ObjectInputStream.java:3129)
        at java.io.ObjectInputStream$BlockDataInputStream.readUTFBody(ObjectInputStream.java:3062)
        at java.io.ObjectInputStream$BlockDataInputStream.readUTF(ObjectInputStream.java:2874)
        at java.io.ObjectInputStream.readString(ObjectInputStream.java:1639)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1342)
        at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2000)
        at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:1924)
        at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:1801)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1351)
        at java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2000)
        at java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:1924)
        at java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:1801)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1351)
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:371)
        at net.openhft.chronicle.hash.serialization.impl.SerializableReader.read(SerializableReader.java:43)

**Comment by RobAustin** [comment]
it looks like the data that it was reading became corrupted. Are you able to provide a reproducible test case? My initial guess, and it is just a guess, it is possible that it is related to where the data was written into the map, and at this stage, I would ignore time-related or concurrency issues. In other words, once it fails if you replay the same records into the map in the same order, does it fail at the same point.

**Comment by mortterna** [comment]
We stumbled on the same problem using latest version (3.17.4). After downgrading to version 3.16.4 the problem went away. We have a reliable way to reproduce the problem but it involves a json-document and jackson deserialization to our java model.

**Comment by samimakicc** [comment]
Two ways to replicate the bug.
First one throws UTFDataFormatException and the second one simply corrupts the value.
```
import net.openhft.chronicle.map.ChronicleMap;
import net.openhft.chronicle.map.ChronicleMapBuilder;

import java.io.Serializable;

public class Test1 {
    public static void main(String[] args) {
        ChronicleMap<Integer, Foo> map = ChronicleMapBuilder.of(Integer.class, Foo.class)
                .name("foo")
                .averageValueSize(2500)
                .entries(10)
                .create();

        System.out.println("put 1");
        map.put(1, new Foo(513));
        System.out.println("get 1");
        map.get(1);
        System.out.println("works");
        System.out.println("put 2");
        map.put(2, new Foo(514));
        System.out.println("get 2");
        map.get(2);
        System.out.println("doesn't work");
    }
}


class Foo implements Serializable {
    String x;

    Foo(int length) {
        StringBuilder sb = new StringBuilder();
        sb.append("x");
        for (int i = 0; i < length; ++i) {
            sb.append("ä");
        }
        this.x = sb.toString();
    }
}
```

```
import net.openhft.chronicle.map.ChronicleMap;
import net.openhft.chronicle.map.ChronicleMapBuilder;

import java.io.Serializable;

public class Test2 {
    public static void main(String[] args) {
        ChronicleMap<Integer, Bar> map = ChronicleMapBuilder.of(Integer.class, Bar.class)
                .name("bar")
                .averageValueSize(2500)
                .entries(10)
                .create();

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < 2000; ++i) {
            sb.append(i % 10);
        }
        String expected = sb.toString();

        map.put(1, new Bar(expected));
        String actual = map.get(1).x;
        
        System.out.println(expected.equals(actual)); // prints false, actual gets corrupted around byte 1023
        System.out.println(expected);
        System.out.println(actual);
    }
}


class Bar implements Serializable {
    String x;

    Bar(String x) {
        this.x = x;
    }
}
```

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.6](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.6), [BOM-2.17.396](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.396)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.6](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.6), [BOM-2.17.407](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.407)
